extensions/VisualEditor/wikia/ApiAddMediaPermanent.php

```diff
@@ -41,7 +41,8 @@ private function executeImage() {
 			$file = new LocalFile( $title, RepoGroup::singleton()->getLocalRepo() );
 			$file->upload( $tempFile->getPath(), '', $pageText ? $pageText : '' );
 			return array(
-				'title' => $file->getTitle()->getText()
+				'title' => $file->getTitle()->getText(),
+				'url' => $file->getUrl()
```

Comment:
API response should be consistent regardless if uploaded image is duplicate or not - so perhaps you should apply similar modification around line 32 of this file. However: this API originally didn't return URL because client should be able to construct URL to thumbnail based on title (and requested size) - if such functionality (of constructing URL) is currently missing in Mercury then I suggest building it for Mercury - it would be more useful and generic than just modifying this one API.